portal-impl/src/com/liferay/portal/verify/PreupgradeVerifyStoreFileSystemStructure.java

```diff
@@ -117,4 +153,196 @@ private Path _getRootDirPath() throws Exception {
+					if (!_hasAdvancedFileSystemStructureRepositoryIdPath(
+							fileNamePath)) {
+
+						return false;
+					}
+				}
+				else if ((fileName.length() > 2) &&
+						 Validator.isNull(FileUtil.getExtension(fileName))) {
+
+					_log.error(
+						StringBundler.concat(
+							"File ", fileNamePath.toString(),
+							" name has more than 2 characters and no ",
+							"extension in advanced file system structure"));
+
+					return false;
+				}
+			}
+
+			return true;
+		}
+	}
+
+	private boolean _hasFileSystemStructureCompanyIdPath(Path companyIdPath)
+		throws IOException {
+
+		try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(
+				companyIdPath)) {
+
+			for (Path repositoryIdPath : directoryStream) {
+				if (_isSystemCompanyRepositoryIdPath(repositoryIdPath)) {
+					continue;
+				}
+
+				if (!Files.isDirectory(repositoryIdPath)) {
+					_log.error(
+						"Unexpected file " + repositoryIdPath +
+							" in file system structure");
+
+					return false;
+				}
+
+				if (!_hasFileSystemStructureRepositoryIdPath(
+						repositoryIdPath)) {
+
+					return false;
+				}
+			}
+
+			return true;
+		}
+	}
+
+	private boolean _hasFileSystemStructureFileNamePath(Path fileNamePath)
+		throws IOException {
+
+		if (StringUtil.contains(
+				String.valueOf(fileNamePath.getFileName()), StringPool.PERIOD,
+				StringPool.BLANK)) {
+
+			_log.error(
+				StringBundler.concat(
+					"Unexpected file name directory ", fileNamePath.toString(),
+					" with extension in file system structure"));
+
+			return false;
+		}
+
+		try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(
+				fileNamePath)) {
+
+			for (Path versionLabelPath : directoryStream) {
+				if (Files.isDirectory(versionLabelPath)) {
+					continue;
+				}
+
+				String versionLabelName = String.valueOf(
+					versionLabelPath.getFileName());
+
+				if (StringUtil.equals(
+						versionLabelName,
+						DLFileEntryConstants.PRIVATE_WORKING_COPY_VERSION)) {
+
+					continue;
+				}
+
+				if (!versionLabelName.matches("\\d+\\.\\d+.*")) {
+					_log.error(
+						"Unexpected file " + versionLabelPath +
+							" not matching version label pattern");
+
+					return false;
+				}
+			}
+
+			return true;
+		}
+	}
+
+	private boolean _hasFileSystemStructureRepositoryIdPath(
+		Path repositoryIdPath) {
+
+		try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(
+				repositoryIdPath)) {
+
+			for (Path fileNamePath : directoryStream) {
+				if (!Files.isDirectory(fileNamePath)) {
+					_log.error(
+						"Unexpected file " + fileNamePath +
+							" in file system structure");
+
+					return false;
+				}
+
+				if (!_hasFileSystemStructureFileNamePath(fileNamePath)) {
+					return false;
+				}
+			}
+
+			return true;
+		}
+		catch (Exception exception) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					"Unable to verify file system structure in: " +
+						repositoryIdPath,
+					exception);
+			}
```

Comment:
@IstvanD I just noticed after forwarding the pull that I forgot to remove this warn block so the IOException is handled at first level (in the same way I did in a51e652cca9d084b6b5f2b14eb5b5c3d9efcd9f9)

Can you remove it and send again?

Sorry for not catching this before